halo2-base/src/utils.rs

```diff
@@ -233,6 +287,10 @@ pub fn decompose_bigint<F: BigPrimeField>(e: &BigInt, num_limbs: usize, bit_len:
     }
 }
 
+/// Decomposes an immutable reference to a `BigInt` into `num_limbs` limbs of `bit_len` bits each and returns a [Vec] of [BigPrimeField] represented by those limbs wrapped in [Value].
+/// * `e`: An immutable reference to `BigInt` to decompose
+/// * `num_limbs`: The number of limbs to decompose `e` into
+/// * `bit_len`: The number of bits in each limb
```

Comment:
Assumes `bit_len < 128`

Also check if we still use this function anywhere? If not maybe we can delete it